pkg/sources/github/github.go

```diff
@@ -770,46 +777,36 @@ func (s *Source) scan(ctx context.Context, installationClient *github.Client, ch
 				return nil
 			}
 
-			logger := s.log.WithValues("repo", repoURL)
-			logger.V(2).Info(fmt.Sprintf("attempting to clone repo %d/%d", i+1, len(s.repos)))
-			var path string
-			var repo *gogit.Repository
-			var err error
-
-			path, repo, err = s.cloneRepo(ctx, repoURL, installationClient)
+			// Scan the repository
+			repoCtx := context.WithValues(ctx, "repo", repoURL)
+			duration, err := s.cloneAndScanRepo(repoCtx, installationClient, repoURL, chunksChan)
 			if err != nil {
 				scanErrs.Add(err)
 				return nil
 			}
+			ctx.Logger().V(2).Info(fmt.Sprintf("scanned %d/%d repos", scanned, len(s.repos)), "duration_seconds", duration)
 
-			defer os.RemoveAll(path)
-			if err != nil {
-				scanErrs.Add(fmt.Errorf("error cloning repo %s: %w", repoURL, err))
-				return nil
-			}
-
-			s.setScanOptions(s.conn.Base, s.conn.Head)
-
-			repoSize := s.repoSizes.getRepo(repoURL)
-			logger.V(2).Info(fmt.Sprintf("scanning repo %d/%d", i, len(s.repos)), "repo_size_kb", repoSize)
+			// Scan the repository's wiki, if it has one.
+			if _, ok := s.repoWikiCache[repoURL]; ok {
+				wikiURL := strings.TrimSuffix(repoURL, ".git") + ".wiki.git"
+				wikiCtx := context.WithValue(ctx, "repo", wikiURL)
 
-			now := time.Now()
-			defer func(start time.Time) {
-				logger.V(2).Info(fmt.Sprintf("scanned %d/%d repos", scanned, len(s.repos)), "repo_size", repoSize, "duration_seconds", time.Since(start).Seconds())
-			}(now)
-
-			if err = s.git.ScanRepo(ctx, repo, path, s.scanOptions, sources.ChanReporter{Ch: chunksChan}); err != nil {
-				scanErrs.Add(fmt.Errorf("error scanning repo %s: %w", repoURL, err))
-				return nil
+				_, err := s.cloneAndScanRepo(wikiCtx, installationClient, wikiURL, chunksChan)
+				if err != nil {
+					scanErrs.Add(err)
+					return nil
```

Comment:
is there any chance that the repo comments are still scannable even if wiki scanning fails?

Reply:
That's a good point.